src/widget.ts

```diff
@@ -1,16 +1,19 @@
-import { DOMWidgetView } from "@jupyter-widgets/base";
+import { DOMWidgetView, DOMWidgetModel } from "@jupyter-widgets/base";
 import { vegaEmbed } from "./index";
 import { Result } from "vega-embed";
+import * as ndarray from "ndarray";
+import { table_serialization, rowProxy, IDict } from "jupyter-tablewidgets";
 
 interface WidgetUpdate {
   key: string;
   remove?: string;
-  insert?: any[];
+  insert?: any[] | string;
```

Comment:
What does it mean to insert a string?

Reply:
Maybe you want to list the exact strings

```suggestion
  insert?: any[] | '@dataframe' | ...;
```